core/business.go

```diff
@@ -26,22 +27,25 @@ type Metrics struct {
 
 // TransferRequest data type
 type TransferRequest struct {
-	TimeStamp int64  `json:"ts"`       // timestamp of the request
-	File      string `json:"file"`     // LFN name to be transferred
-	Block     string `json:"block"`    // block name to be transferred
-	Dataset   string `json:"dataset"`  // dataset name to be transferred
-	SrcUrl    string `json:"srcUrl"`   // source agent URL which initiate the transfer
-	SrcAlias  string `json:"srcAlias"` // source agent name
-	DstUrl    string `json:"dstUrl"`   // destination agent URL which will consume the transfer
-	DstAlias  string `json:"dstAlias"` // destination agent name
-	Delay     int    `json:"delay"`    // transfer delay time, i.e. post-pone transfer
+	TimeStamp     int64          `json:"ts"`       // timestamp of the request
+	File          string         `json:"file"`     // LFN name to be transferred
+	Block         string         `json:"block"`    // block name to be transferred
+	Dataset       string         `json:"dataset"`  // dataset name to be transferred
+	SrcUrl        string         `json:"srcUrl"`   // source agent URL which initiate the transfer
+	SrcAlias      string         `json:"srcAlias"` // source agent name
+	DstUrl        string         `json:"dstUrl"`   // destination agent URL which will consume the transfer
+	DstAlias      string         `json:"dstAlias"` // destination agent name
+	Delay         int            `json:"delay"`    // transfer delay time, i.e. post-pone transfer
+	Id            int64          `json:"id"`       // unique id of each request
+	Priority      int            `json:"priority"` // priority of request
+	Status				string				 `json:"status"`	 // Identify the category of request
+	FailedRecords []CatalogEntry // Store records which are failed
```

Comment:
I added FailedRecords parameter to store the failed transfer data. This will help us to separate out the failed and succeeded data after first and second iterations.